arcgis-ios-sdk-samples/Analysis/Viewshed (GeoElement)/ViewshedGeoElementViewController.swift

```diff
@@ -0,0 +1,136 @@
+        scene.baseSurface = surface
+
+        // add a scene layer
+        let buildings = AGSArcGISSceneLayer(url: URL(string:"http://tiles.arcgis.com/tiles/P3ePLMYs2RVChkJx/arcgis/rest/services/Buildings_Brest/SceneServer/layers/0")!)
+        scene.operationalLayers.add(buildings)
+
+        // create a graphics overlay for the tank
+        let graphicsOverlay = AGSGraphicsOverlay()
+        graphicsOverlay.sceneProperties = AGSLayerSceneProperties(surfacePlacement: .relative)
+        sceneView.graphicsOverlays.add(graphicsOverlay)
+        
+        // set up heading expression for tank
+        let renderer3D = AGSSimpleRenderer()
+        let sceneProperties = AGSRendererSceneProperties(headingExpression: "[heading] + 90", pitchExpression: "[pitch]", rollExpression: "[roll]")
+        sceneProperties.headingExpression = "[HEADING]"
+        renderer3D.sceneProperties = sceneProperties
+        graphicsOverlay.renderer = renderer3D
+
+        // create a graphic of a tank
+        let tankSymbol = AGSModelSceneSymbol(name: "bradle", extension: "3ds", scale: 10.0)
+        tankSymbol.heading = 90.0
+        tankSymbol.anchorPosition = .bottom
+        tank = AGSGraphic(geometry: AGSPoint(x: -4.506390, y: 48.385624, spatialReference: AGSSpatialReference.wgs84()),
+                          symbol: tankSymbol,
+                          attributes: ["HEADING": 0.0])
+        graphicsOverlay.graphics.add(tank)
+        
+        // create a viewshed to attach to the tank
+        let geoElementViewshed = AGSGeoElementViewshed(geoElement: tank,
+                                                       horizontalAngle: 90.0,
+                                                       verticalAngle: 40.0,
+                                                       minDistance: 0.1,
+                                                       maxDistance: 250.0,
+                                                       headingOffset: 0.0,
+                                                       pitchOffset: 0.0)
+        
+        // offset viewshed observer location to top of tank
+        geoElementViewshed.offsetZ = 3.0
+        
+        // create an analysis overlay to add the viewshed to the scene view
+        let analysisOverlay = AGSAnalysisOverlay()
+        analysisOverlay.analyses.add(geoElementViewshed)
+        sceneView.analysisOverlays.add(analysisOverlay)
+
+        // set camera controller to follow tank
+        let cameraController = AGSOrbitGeoElementCameraController(targetGeoElement: tank, distance: 200.0)
+        cameraController.cameraPitchOffset = 45.0
+        sceneView.cameraController = cameraController
+        
+        sceneView.scene = scene
+    }
+
+    func geoView(_ geoView: AGSGeoView, didTapAtScreenPoint screenPoint: CGPoint, mapPoint: AGSPoint) {
+        // set the new waypoint
+        waypoint = mapPoint;
+        
+        // start a new timer
+        timer = Timer.scheduledTimer(timeInterval: 0.1,
```

Comment:
can we elaborate on the comment to say " start a timer to animate towards the waypoint"